src/storage/src/table/mod.rs

```diff
@@ -40,20 +38,21 @@ pub trait TableIter: Send {
     async fn next_row(&mut self) -> StorageResult<Option<OwnedRow>>;
 }
 
-pub async fn collect_data_chunk<E, S>(
+pub async fn collect_data_chunk<E, S, R>(
     stream: &mut S,
     schema: &Schema,
     chunk_size: Option<usize>,
 ) -> Result<Option<DataChunk>, E>
 where
-    S: Stream<Item = Result<KeyedRow<Bytes>, E>> + Unpin,
+    S: Stream<Item = Result<R, E>> + Unpin,
+    R: Row,
 {
     let mut builders = schema.create_array_builders(chunk_size.unwrap_or(0));
     let mut row_count = 0;
     for _ in 0..chunk_size.unwrap_or(usize::MAX) {
         match stream.next().await.transpose()? {
             Some(row) => {
-                for (datum, builder) in row.iter().zip_eq_fast(builders.iter_mut()) {
+                for (datum, builder) in row.exact_size_iter().zip_eq_fast(builders.iter_mut()) {
```

Comment:
...or switch to `zip_eq_debug` instead. There's no much difference. 🤣

Reply:
I have switched to use `zip_eq_debug`.